commit/chainfee/processor.go

```diff
@@ -3,41 +3,134 @@ package chainfee
 import (
 	"context"
 	"fmt"
+	"math/big"
+	"sort"
+	"time"
 
 	mapset "github.com/deckarep/golang-set/v2"
+	commonconfig "github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"golang.org/x/exp/maps"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon/consensus"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	readerpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
 
 	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
 )
 
-type Processor struct {
+type processor struct {
+	destChain                        cciptypes.ChainSelector
+	lggr                             logger.Logger
+	homeChain                        reader.HomeChain
+	ccipReader                       readerpkg.CCIPReader
+	ChainFeePriceBatchWriteFrequency commonconfig.Duration
+	fRoleDON                         int
 }
 
-func NewProcessor() *Processor {
-	return &Processor{}
+// nolint: revive
+func NewProcessor(
+	lggr logger.Logger,
+	destChain cciptypes.ChainSelector,
+	homeChain reader.HomeChain,
+	ccipReader readerpkg.CCIPReader,
+	chainFeePriceBatchWriteFrequency commonconfig.Duration,
+	fRoleDON int,
+) *processor {
```

Comment:
Either: `processor` should be public, or this should return `PluginProcessor[blah, blah, blah]`